src/rpcprotocol.cpp

```diff
@@ -287,3 +289,68 @@ UniValue JSONRPCError(int code, const string& message)
     error.push_back(Pair("message", message));
     return error;
 }
+
+/** Username used when cookie authentication is in use (arbitrary, only for
```

Comment:
I think `rpcprotocol.h/.cpp` is a good place for this extension. A new `cookieauth.h/cpp` file would be possible but i think is more compact and neatly arranged if it stays in `rpcprotocol.h/cpp`